types/lodash/common/collection.d.ts

```diff
@@ -209,8 +209,9 @@ declare module "../index" {
         filter<T, S extends T>(collection: List<T> | null | undefined, predicate: ListIteratorTypeGuard<T, S>): S[];
         /**
          * @see _.filter
-         */
-        filter<T>(collection: List<T> | null | undefined, predicate?: ListIterateeCustom<T, boolean>): T[];
+        */
+       filter<T>(collection: List<T> | null | undefined): Array<Exclude<T, null | undefined>>;
```

Comment:
Nit: can you add false literal types as well? (false, 0, '')

Reply:
I even found the `Falsey` union type provided by lodash for that, done.